CoreDistributedCache/NHibernate.Caches.CoreDistributedCache.SqlServer/SqlServerFactory.cs

```diff
@@ -0,0 +1,112 @@
+		public SqlServerFactory(
+			string connectionString, string schemaName, string tableName, TimeSpan? expiredItemsDeletionInterval)
+		{
+			_options = new SqlServerCacheOptions
+			{
+				ConnectionString = connectionString,
+				SchemaName = schemaName,
+				TableName = tableName,
+				ExpiredItemsDeletionInterval = expiredItemsDeletionInterval
+			};
+		}
+
+		/// <summary>
+		/// Constructor with configuration properties. It supports <c>connection-string</c>, <c>schema-name</c>,
+		/// <c>table-name</c> and <c>expired-items-deletion-interval</c> properties.
+		/// See <see cref="SqlServerCacheOptions" />.
+		/// </summary>
+		/// <param name="properties">The configurations properties.</param>
+		/// <remarks>
+		/// If <c>expired-items-deletion-interval</c> is provided as an integer, this integer will be used as a number
+		/// of minutes. Otherwise the setting will be parsed as a <see cref="TimeSpan" />.
+		/// </remarks>
+		public SqlServerFactory(IDictionary<string, string> properties)
+		{
+			_options = new SqlServerCacheOptions();
+
+			if (properties == null)
+				return;
+
+			if (properties.TryGetValue(_connectionString, out var connectionString))
+			{
+				_options.ConnectionString = connectionString;
+				Log.Info("ConnectionString set as '{0}'", connectionString);
+			}
+			else
+				Log.Warn("No {0} property provided", _connectionString);
+
+			if (properties.TryGetValue(_schemaName, out var schemaName))
+			{
+				_options.SchemaName = schemaName;
+				Log.Info("SchemaName set as '{0}'", schemaName);
+			}
+			else
+				Log.Warn("No {0} property provided", _schemaName);
+
+			if (properties.TryGetValue(_tableName, out var tableName))
+			{
+				_options.TableName = tableName;
+				Log.Info("TableName set as '{0}'", tableName);
+			}
+			else
+				Log.Warn("No {0} property provided", _tableName);
+
+			if (properties.TryGetValue(_expiredItemsDeletionInterval, out var eidi))
+			{
+				if (eidi != null)
+				{
+					if (int.TryParse(eidi, out var minutes))
+						_options.ExpiredItemsDeletionInterval = TimeSpan.FromMinutes(minutes);
+					else if (TimeSpan.TryParse(eidi, out var expirationScanFrequency))
+						_options.ExpiredItemsDeletionInterval = expirationScanFrequency;
+					else
+						Log.Warn(
+							"Invalid value '{0}' for {1} setting: it is neither an int nor a TimeSpan. Ignoring.",
+							eidi, _expiredItemsDeletionInterval);
+				}
+				else
+				{
+					Log.Warn("Invalid property {0}: it lacks a value. Ignoring.", _expiredItemsDeletionInterval);
+				}
+			}
+		}
+
+		/// <inheritdoc />
+		public CacheConstraints Constraints { get; } = new CacheConstraints { MaxKeySize = 449 };
+
+		/// <inheritdoc />
+		public IDistributedCache BuildCache()
+		{
+			// According to https://docs.microsoft.com/en-us/aspnet/core/performance/caching/distributed#the-idistributedcache-interface
+			// (see its paragraph end note) there is no need for a singleton lifetime.
+			return new SqlServerCache(_options);
```

Comment:
Same question than for `RedisCache`, but with a different concern: each `SqlServerCache` may spawn its own background task for deleting expired items, at each operation on itself, if enough time has elapsed since the last time the instance have done it. So the more instances we create, the more such background task may be run concurrently. Moreover this causes the expired items deletion to occur overall much more frequently than asked through `ExpiredItemsDeletionInterval`.